modules/payloads/singles/cmd/unix/pingback_bind.rb

```diff
@@ -0,0 +1,57 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core/handler/bind_tcp'
+require 'msf/core/payload/pingback'
+require 'msf/base/sessions/pingback'
+require 'msf/base/sessions/pingback_options'
+
+module MetasploitModule
+
+  CachedSize = 90
+
+  include Msf::Payload::Single
+  include Msf::Sessions::PingbackOptions
+
+  def initialize(info = {})
+    super(merge_info(info,
+      'Name'          => 'Unix Command Shell, Pingback Bind TCP (via netcat)',
+      'Description'   => 'Accept a connection, send a UUID, then exit',
+      'Author'         =>
+        [
+          'asoto-r7'
+        ],
+      'License'       => MSF_LICENSE,
+      'Platform'      => 'unix',
+      'Arch'          => ARCH_CMD,
+      'Handler'       => Msf::Handler::BindTcp,
+      'Session'       => Msf::Sessions::Pingback,
+      'PayloadType'   => 'cmd',
+      'RequiredCmd'   => 'netcat',
+      'Payload'       =>
+        {
+          'Offsets' => { },
+          'Payload' => ''
+        }
+      ))
+  end
+
+  #
+  # Constructs the payload
+  #
+  def generate
+    return super + command_string
+  end
+
+  #
+  # Returns the command string to use for execution
+  #
+  def command_string
+    pingback_uuid ||= generate_pingback_uuid
+    pingback_uuid.gsub!('-','')
+
+    "echo -ne '#{pingback_uuid.scan(/../).map { |x| "\\x" + x }.join}' | nc -lp #{datastore['LPORT']}"
```

Comment:
`echo -ne` is not portable. Ideally, we'd want to make `Msf::Post::File#_write_file_unix_shell` more generic and return a command, then reuse the code throughout Framework.

https://github.com/rapid7/metasploit-framework/blob/026b38eb71cba98b88bf860c1fc987ea5a6eda37/lib/msf/core/post/file.rb#L508-L638

Reply:
(EDIT: Github is being weird.)

Oh, good point.  I'm uneasy about modifying that method, since it's focused on writing a file and I suspect some far-reaching and difficult-to-test implications.  The quick solution would be to use `printf` instead of `echo`:

```
  "printf '#{pingback_uuid.scan(/../).map { |x| "\\x" + x }.join}' | nc -lp #{datastore['LPORT']}"
```

I've added that change via the latest commit: 928e4679ae45847bda3bce3303001c842ced79cc

Reply:
Sounds good! We can always refactor as desired later. Thanks.